### PR TITLE
Celestia publish

### PR DESCRIPTION
**Pull request by agouin**
Enables publishing block data to Celestia via the proposer

### All Validators

Configure Celestia Address in keyring for posting blocks

```bash
rcd keys tiablob add
# OR
rcd keys tiablob restore "some mnemonic"
```

# Celestia blob payment methods

## 1: Validator funded
Validators fund their own Celestia addresses for payment

1. Fund Validator Celestia Addresses
https://docs.celestia.org/nodes/mocha-testnet#mocha-testnet-faucet

## 2: Off-chain feegrants
A centrally controlled account fee-grants validator payment accounts

### Centrally controlled account instructions

1. Configure Celestia Address in keyring for feegranting validators.
```bash
rcd keys tiablob add --feegrant
# OR
rcd keys tiablob restore --feegrant "some mnemonic"
```

2. Fund Celestia Feegranter Address
https://docs.celestia.org/nodes/mocha-testnet#mocha-testnet-faucet

3. Enable feegrants for all validators
```bash
rcd tx tiablob feegrant [validator1CelestiaAddress] [validator2CelestiaAddress] ...
```

### Validator instructions

1. Add feegranter address to keyring
```bash
rcd keys tiablob add --feegrant --address celestia1dr3gwf5kulm4e4k0pctwzn0htw6wrvevdgjdlf
```

## 3. Feegrants controlled by ICA

TODO in follow up PR

# Start up chain

Blocks will be published to Celestia as the chain progresses.

NOTE: Until proving loop is implemented, we do not have confirmation that blocks have been published.

### CODE REVIEW

File: simapp/cmd/rcd/commands.go [hunk 1]
```diff
@@ -81,18 +86,47 @@ func initAppConfig() (string, interface{}) {
...
+			GasAdjustment:      1.0,
+			ProofQueryInterval: 12 * time.Second,
+			MaxFlushSize:       32,
+		},
 	}
 
-	customAppTemplate := serverconfig.DefaultConfigTemplate
+	customAppTemplate := serverconfig.DefaultConfigTemplate + `
+
+	[celestia]
```
Comment (misko9) [inline simapp/cmd/rcd/commands.go]: Should the tiablob default config template live in the module instead of simapp?

File: relayer/publish.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	if height <= 1 {
+		return
+	}
+
+	// only publish every n blocks
+	if (height-1)%int64(n) != 0 {
+		return
+	}
+
+	if !r.provider.KeyExists(CelestiaPublishKeyName) {
```
Comment (misko9) [inline relayer/publish.go:38]: Can we do this check earlier, e.g. in NewRelayer()?

**Review by misko9 — COMMENTED** [review]
Looks great!

File: simapp/cmd/rcd/commands.go (hunk 1 above)
Comment (agouin) [reply simapp/cmd/rcd/commands.go]: yes, ty.

File: relayer/publish.go (hunk 2 above)
Comment (agouin) [reply relayer/publish.go:38]: we want to check every time so that the operator can manage their keys while the process is running.

**Pull request merged**